### PR TITLE
Add MultiFab::sum(region) and sum_unique(region)

### PR DESCRIPTION
**Pull request by WeiqunZhang**
## Summary

Add two new functions to MultiFab that return the sum of the given region.

## Additional background

#3869 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by asalmgren** [comment]
1) duplicatly is misspelled
2) I'm worried about people assuming this works correctly with periodicity even though the code has a comment that it doesn't account for that

**Comment by WeiqunZhang** [comment]
Yes, there could be potential confusion even though logically speaking it should not because without being provided the periodicity information amrex cannot know how to handle it. So the caller has to assume periodicity is not part of the consideration. This is a feature request from WarpX, which does not need the periodic boundaries to be handled in this function. I didn't try to add periodicity support because it's actually quite awkward to implement. We would need to memcpy a vector of periodically shifted Boxes to device in order to do all these in one kernel. Maybe we can wait till someone wants the feature.

File: Src/Base/AMReX_MultiFab.H [hunk 1]
```diff
@@ -249,6 +253,13 @@ public:
                      bool local = false,
                      const Periodicity& period = Periodicity::NonPeriodic()) const;
     /**
+    * \brief Returns the unique sum of component "comp" in the given
+    * region. Non-unique points owned by multiple boxes in the MultiFab are
+    * skipped. No ghost cells are included. This function does not take
```
Comment (ax3l) [inline Src/Base/AMReX_MultiFab.H]:
```suggestion
    * region. Non-unique points owned by multiple boxes in the MultiFab are
    * only added once. No ghost cells are included. This function does not take
```

Comment (ax3l) [reply Src/Base/AMReX_MultiFab.H]: (maybe same update to the doc string above of the original `sum_unique`)

**Review by ax3l — APPROVED** [review]
Thank you! :)

**Pull request merged**